src/localize/LocaleMacro.hx

```diff
@@ -43,8 +47,16 @@ class LocaleMacro {
             pos: Context.currentPos(),
             kind: FFun({
                 ret: null,
-                args: [{name: "catalog", type: macro : Catalog}],
-                expr: macro super(catalog)
+                args: [
+                    {name: "catalog", type: macro : Catalog},
+                    {name: "pos", type: macro : haxe.PosInfos, opt: true},
+                ],
+                expr: macro {
+                    // TODO: can we do the checking at compile-time?
```

Comment:
I understand that. That's why I wrote down that note.
But the compile-time check is rather non-trivial so it is out of the scope of this PR.
For now I will just make the run-time check an opt-in with the define `hxlocalize_runtime_check`.